src/controllers/CheckYourAppealController.ts

```diff
@@ -26,7 +34,38 @@ export class CheckYourAppealController extends BaseAsyncHttpController {
             .orDefault({});
 
         return this.render('check-your-appeal', { ...appealsData, userProfile });
+    }
+
+    @httpPost('')
+    public async handleFormSubmission(req: Request): Promise<HttpResponseMessage> {
+        const userProfile = req.session
+            .chain(_ => _.getValue<ISignInInfo>(SessionKey.SignInInfo))
+            .map(info => info[SignInInfoKeys.UserProfile])
+            .extract() as IUserProfile;
+
+        const appealsData = req.session
+            .chain(_ => _.getExtraData())
+            .chain(data => Maybe.fromNullable(data[AppealKeys.APPEALS_KEY]))
+            .extract() as Appeal;
 
+        try {
+            await this.emailService.send({
```

Comment:
do we need to check first that the userProfile/appealsData is not `undefined` before sending the email?

Reply:
Valid concern. That you for raising it.

I did not want to do that extra check in the scope of the story. I should really expect data to be there on submission otherwise check your answers page should not even display. I expected that there will be some validation of session state when user visits CheckYourAppealController for the first time. Not sure whether it was missed when we implemented other story or whether we created other story. Anyway we need to address it, ideally as separate ticket or at least PR.

Reply:
I agree with this